src/frequenz/sdk/actor/data_sourcing/data_sourcing.py

```diff
@@ -0,0 +1,79 @@
+"""The DataSourcing Actor.
+
+Copyright
+Copyright © 2022 Frequenz Energy-as-a-Service GmbH
+
+License
+MIT
+"""
+
+import logging
+from typing import Dict, Optional
+
+from frequenz.channels import Receiver
+
+from frequenz.sdk.actor import actor
+from frequenz.sdk.actor.channel_registry import ChannelRegistry
+from frequenz.sdk.data_pipeline import ComponentMetricRequest
+from frequenz.sdk.microgrid import ComponentCategory, microgrid_api
+
+from .microgrid_api_source import MicrogridApiSource
+
+
+@actor
+class DataSourcingActor:
+    """An actor that provides data streams of metrics as time series."""
+
+    def __init__(
+        self,
+        request_receiver: Receiver[ComponentMetricRequest],
+        registry: ChannelRegistry,
```

Comment:
we probably need a global singleton registry as well,  before this can be merged.

Reply:
Why? I mean why it is a requirement for this to be merged?

Reply:
Because we're passing the registry as an argument to an actor,  and that's not ideal.

We can of course merge,  but then we'll have to come back and fix this afterwards.